qemu/tests/qemu_guest_agent.py

```diff
@@ -1828,6 +1830,97 @@ def gagent_check_nonexistent_cmd(self, test, params, env):
+            cmd = " && ".join(cmd_list)
+            return " && ".join([cmd_header, cmd, cmd_footer])
+
+        session = self._get_session(params, self.vm)
+        image_size_stg0 = params["image_size_stg0"]
+
+        error_context.context("Format the new data disk and mount it.", logging.info)
+        if params.get("os_type") == "linux":
+            disk_data = list(utils_disk.get_linux_disks(session).keys())
+            mnt_point = utils_disk.configure_empty_disk(
+                session, disk_data[0], image_size_stg0, "linux", labeltype="msdos")
+            partition = "/dev/%s" % mnt_point[0].split("/")[-1]
+        else:
+            disk_index = utils_misc.wait_for(
+                lambda: utils_disk.get_windows_disks_index(session, image_size_stg0), 120)
+            if disk_index:
+                logging.info("Clear readonly for disk and online it in windows guest.")
+                if not utils_disk.update_windows_disk_attributes(session, disk_index):
+                    test.error("Failed to update windows disk attributes.")
+                mnt_point = utils_disk.configure_empty_disk(
+                    session, disk_index[0], image_size_stg0, "windows", labeltype="msdos")
+        error_context.context("Freeze fs.", logging.info)
+        self.gagent.fsfreeze()
+
+        error_context.context("Umount fs or offline disk in guest.", logging.info)
+        if params.get("os_type") == "linux":
+            if not utils_disk.umount(partition, mnt_point[0], session=session):
+                test.fail("Can't umount fs with fsfreeze.")
+        else:
+            detail_cmd = ' echo detail disk'
+            detail_cmd = wrap_windows_cmd(detail_cmd)
+            offline_cmd = ' echo offline disk'
+            offline_cmd = wrap_windows_cmd(offline_cmd)
+            did = disk_index[0]
+            logging.info("Detail for 'Disk%s'" % did)
+            details = session.cmd_output(detail_cmd % did)
+            if re.search("Status.*Online", details, re.I | re.M):
+                logging.info("Offline 'Disk%s'" % did)
+                status, output = session.cmd_status_output(offline_cmd % did,
+                                                           timeout=120)
+                if status != 0:
+                    test.fail("Can not offline disk: %s with fsfreeze." % output)
+        error_context.context("Thaw fs.", logging.info)
+        try:
+            self.gagent.fsthaw()
+        except guest_agent.VAgentCmdError as detail:
+            if not re.search("fsfreeze is limited up to 10 seconds", str(detail)):
+                test.error("guest-fsfreeze-thaw cmd failed with: ('%s')"
+                           % str(detail))
+        error_context.context("Mount fs or online disk in guest.", logging.info)
+        if params.get("os_type") == "linux":
+            if not utils_disk.mount(partition, mnt_point[0], session=session):
+                test.fail("Can't mount fs with fsthaw.")
```

Comment:
@xiagao For different guest. It's behavior is different.  e.g  RHEL6 it should fail. RHEL7 and RHEL8 it should pass.